@here/harp-lines/lib/Lines.ts

```diff
@@ -27,92 +27,70 @@ interface VertexAttributeDescriptor {
 /**
  * Declares all the vertex attributes used for rendering a line using the [[SolidLineMaterial]].
  */
-const LINE_VERTEX_ATTRIBUTES: VertexAttributeDescriptor[][] = [
-    [
-        { name: "texcoord", itemSize: 2, offset: 0 },
-        { name: "position", itemSize: 2, offset: 2 },
-        { name: "bitangent", itemSize: 3, offset: 4 }
-    ],
-    [
-        { name: "texcoord", itemSize: 2, offset: 0 },
-        { name: "position", itemSize: 3, offset: 2 },
-        { name: "bitangent", itemSize: 4, offset: 5 }
-    ]
+const LINE_VERTEX_ATTRIBUTES: VertexAttributeDescriptor[] = [
+    { name: "texcoord", itemSize: 2, offset: 0 },
+    { name: "position", itemSize: 3, offset: 2 },
+    { name: "bitangent", itemSize: 4, offset: 5 }
 ];
 
 /** Stride size for line vertex data. */
-const LINE_STRIDE = [7, 9];
+const LINE_STRIDE = 9;
 
 /**
  * Declares all the vertex attributes used for rendering a line using the
  * [[HighPrecisionLineMaterial]].
  */
-const HP_LINE_VERTEX_ATTRIBUTES: VertexAttributeDescriptor[][] = [
-    [
-        { name: "texcoord", itemSize: 2, offset: 0 },
-        { name: "position", itemSize: 2, offset: 2 },
-        { name: "positionLow", itemSize: 2, offset: 4 },
-        { name: "bitangent", itemSize: 3, offset: 6 }
-    ],
-    [
-        { name: "texcoord", itemSize: 2, offset: 0 },
-        { name: "position", itemSize: 3, offset: 2 },
-        { name: "positionLow", itemSize: 3, offset: 5 },
-        { name: "bitangent", itemSize: 4, offset: 8 }
-    ]
+const HP_LINE_VERTEX_ATTRIBUTES: VertexAttributeDescriptor[] = [
+    { name: "texcoord", itemSize: 2, offset: 0 },
+    { name: "position", itemSize: 3, offset: 2 },
+    { name: "positionLow", itemSize: 3, offset: 5 },
+    { name: "bitangent", itemSize: 4, offset: 8 }
 ];
 
 /** Stride size for high precision line vertex data. */
-const HP_LINE_STRIDE = [9, 12];
+const HP_LINE_STRIDE = 12;
 
 /**
  * Class that holds the vertex and index attributes for a [[Lines]] object.
  */
 export class LineGeometry {
-    readonly dimensions: 2 | 3;
     vertices: number[] = [];
     indices: number[] = [];
-
-    constructor(params: { dimensions: 2 | 3 }) {
-        this.dimensions = params.dimensions;
-    }
 }
 
 /**
  * Creates a [[LineGeometry]] object out of a polyline.
  *
  * @param polyline Array of `numbers` describing a polyline.
+ * @param projection [[Projection]] to which project the polyline.
```

Comment:
Passing the whole projection is a bit misleading b/c it looks like this method will project the line but it doesn't. The polyline is already projected (as far as I can see).
Passing a boolean flag would make things more clear.
Please also add some comment that polyline is always 3 dimensional now

Reply:
It's true that currently only the ``getDimensions`` method of the projection is used, but in the future this might not be the case and you might want to do projection-specific computations (i.e. sphere and elliptical projections are not gonna compute the vertex normal in the same way).

Don't mind changing it, but I think it's more future proof this way.